pytest_fixtures/component/maintain.py

```diff
@@ -22,15 +23,36 @@ def module_stash(request):
 
 
 @pytest.fixture(scope='module')
-def sat_maintain(request, module_target_sat, module_capsule_configured):
+def module_satellite_iop(request, module_satellite_host):
+    """Deploy and return Insights on Prem Satellite configured."""
+    satellite_iop = module_satellite_host()
+    satellite_iop.register_to_cdn()
+    satellite_iop.configure_insights_on_prem()
+    return satellite_iop
+
+
+@pytest.fixture(scope='module')
+def module_satellite_maintain(request, module_target_sat):
+    """Provides Satellite for satellite-maintain tests"""
     if settings.remotedb.server:
-        sat = Satellite(settings.remotedb.server)
-        sat.enable_satellite_ipv6_http_proxy()
-        yield sat
-    else:
-        module_target_sat.register_to_cdn(pool_ids=settings.subscription.fm_rhn_poolid.split())
-        hosts = {'satellite': module_target_sat, 'capsule': module_capsule_configured}
-        yield hosts[request.param]
+        logger.info("Using Satellite with remotedb server")
```

Comment:
I know its bit out of scope, but we should use the name what we document
```suggestion
        logger.info('Using Satellite with external DB server')
```